common/txmgr/confirmer.go

```diff
@@ -1105,7 +1105,11 @@ func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) Res
 
 		ec.lggr.Debugw("Callback: resuming tx with receipt", "output", output, "taskErr", taskErr, "pipelineTaskRunID", data.ID)
 		if err := ec.resumeCallback(data.ID, output, taskErr); err != nil {
-			return err
+			return fmt.Errorf("ResumePendingTaskRuns failed: %w", err)
+		}
+		// Mark tx as having completed callback
+		if err := ec.txStore.UpdateTxCallbackCompleted(ctx, data.ID, ec.chainID); err != nil {
+			return fmt.Errorf("ResumePendingTaskRuns failed: %w", err)
```

Comment:
Why don't we add some context about the particular calls that failed? The caller of `ResumePendingTaskRuns` already knows the method failed generally based on the error returned.
```suggestion
			return fmt.Errorf("failed to resume calback: %w", err)
		}
		// Mark tx as having completed callback
		if err := ec.txStore.UpdateTxCallbackCompleted(ctx, data.ID, ec.chainID); err != nil {
			return fmt.Errorf("failed to update callback complete: %w", err)
```

Reply:
Since this is wrapping an error from a more inner function, I'm worried I might repeat the same error message. I was under the impression we had this pattern to wrap the error with the  function name to create a sort of error stack when this bubbles up to the top.

Reply:
We are trying to create a stack, but it should be human readable and differentiate between the cases. IMHO including function names in any way is an anti-pattern. We should prefer to add context that the caller cannot already infer themselves (they already know they called `ResumePendingTaskRuns`), and we should also prefer not to encode implementation details inside of human facing errors (when someone refactors `ResumePendingTaskRuns`, they have to notice all of these strings and update them).

Reply:
Ok think I agree we shouldn't use function names explicitly. But I feel the only context I should include here is something like `resuming pending task runs failed` since the more specific context would be taken care of the underlying error since we're just passing it up here.

Reply:
I don't think that message adds any value for calling code which is already well aware of what func was called. What the calling code doesn't know is which of these particular steps failed.

Reply:
I might be unnecessarily worrying about redundancy in the error message. I was thinking the calling code would know the step it failed on based on the error returned by the inner method. Like if the `UpdateTxCallbackCompleted` method returned a wrapped error with a message such as `failed to update callback complete`, it would just duplicate the message if I wrapped the error with a similar message in `ResumePendingTaskRuns`. But I guess there could be any number of specific errors underneath so this is alright. Sorry for the back and forth on something that's so simple

Reply:
If we are calling something that returns a message that we don't have anything to add to, then it is OK to return the error raw. :shrug: 

> Sorry for the back and forth on something that's so simple

No problem. It seems simpler than it is and there are definitely edge cases.

Reply:
Good point. I've updated errors to include more context and returning them raw where there isn't any more to add.